jupyterhub/tests/mocking.py

```diff
@@ -93,6 +93,9 @@ def user_env(self, env):
     def _cmd_default(self):
         return [sys.executable, '-m', 'jupyterhub.tests.mocksu']
 
+    def delete_forever(self):
```

Comment:
Add a docstring explaining what this is an when it is called:

> Called when a user is deleted. This can do things like request removal of resources such as persistent storage. Only called on stopped spawners, and is likely the last action ever taken for the user. Will only be called once on the user's default Spawner.

Reply:
added in https://github.com/jupyterhub/kubespawner/pull/475/files. Hope that's where you wanted it

Reply:
also added a docstring here in mocking.py